### ISSUE TITLE
Failing during restore_issue

### ISSUE THREAD

**Issue by InfernalWisdom** [AlmaLinux/almalinux-deploy#41]
On a base CentOS 8.3 conversion.  The script completes the distro_sync and the next call is restore_issue.  During this call, the script fails.
-----
Complete!
Run dnf distro-sync -y                                                OK
-----

I have reinstalled CentOS 8.3 several times and have the same issue.  I cannot determine how the restore_issue function would cause the script to stop working at that point.

**Comment by titan-dn** [comment]
I've tested conversion 3 times:
- with default /etc/issue*
- with changed /etc/issue*
- with deleted /etc/issue*
all cases have completed done.

Could you present more information: are they exists on file system, content of this files, etc.
You can add some debug commands (e.g echo) to the script to get more info during execiting.

**Comment by InfernalWisdom** [comment]
Adding the following debug code into the script makes the script complete without errors

# Restore /etc/issue* files
restore_issue() {
    echo -n "In restore_issue function ... "
    for file in /etc/issue /etc/issue.net; do
        [ -f "${file}.bak" ] && mv ${file}.bak ${file}
    done
    echo "done"
}

Complete!
Run dnf distro-sync -y                                                OK
In restore_issue function ... done
Generating grub configuration file ...
done

Migration to AlmaLinux is completed



Running the script without changing anything it fails every time.
Complete!
Run dnf distro-sync -y                                                OK
[root@VM-eb0092 ~]# echo $?
1


## both issue and issue.net contain the same data. 
[root@VM-eb0092 ~]# more /etc/issue
\S
Kernel \r on an \m

**Comment by ngyuki** [comment]
In `backup_issue`, `/etc/issue` and `/etc/issue.net` that have changed are saved to `.bak`.

```sh
backup_issue() {
    for file in $(rpm -Vf /etc/issue | cut -d' ' -f4); do
        if [[ ${file} =~ "/etc/issue" ]]; then
            cp "${file}" "${file}.bak"
        fi
    done
}
```

If file is not modified, `/etc/issue.bak` and `/etc/issue.net.bak` will not be created.

And in `restore_issue`, `/etc/issue.bak` and `/etc/issue.net.bak` are restored.

```sh
restore_issue() {
    for file in /etc/issue /etc/issue.net; do
        [ -f "${file}.bak" ] && mv ${file}.bak ${file}
    done
}
```

However, this way of writing will cause the function to exit non-zero if `/etc/issue.net.bak` does not exist.
Then, the script will be terminated because `set -euo pipefail` is written at the beginning of the script.

The correct answer is probably as follows.

```sh
restore_issue() {
    for file in /etc/issue /etc/issue.net; do
        if [ -f "${file}.bak" ]; then
            mv ${file}.bak ${file}
        fi
    done
}
```

**Comment by Erulezz** [comment]
Same issue here on latest CentOS 8. Latest line I see is the dnf distro sync and the script then terminates.

**Comment by ezamriy** [comment]
Fixed in #42
